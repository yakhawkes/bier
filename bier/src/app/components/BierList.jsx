const React = require('react')
const PropTypes = require('prop-types')
const BierCard = require('./BierCard.jsx')

function BierList(props) {
  const biere = props.biere
  return (
    <div className="bierlist">
      {biere.map(bier =>
        (
          <div key={bier.id}>
            <BierCard bier={bier} />
          </div>
        ),
      )}
    </div>
  )
}
module.exports = BierList

BierList.propTypes = {
  biere: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string,
      name: PropTypes.string,
    }),
  ).isRequired,
}
