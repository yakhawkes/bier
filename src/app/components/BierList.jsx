const React = require('react')

function BierList(props) {
  return (
    <table>
      <tr>
        <th>Name</th>
      </tr>
      {props.biers.map(bier =>
        (
          <tr key={bier.id}>
            <td>{bier.name}</td>
          </tr>
        ),
      )}
    </table>
  )
}
module.exports = BierList
