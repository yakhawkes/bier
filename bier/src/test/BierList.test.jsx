import React from 'react'
import renderer from 'react-test-renderer'
import BierList from '../app/components/BierList.jsx'

const biere = require('./beerspayload.json')

test('List of Biers', () => {
  const component = renderer.create(
    <BierList biere={biere.data} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
