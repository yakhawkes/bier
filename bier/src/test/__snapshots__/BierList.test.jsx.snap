// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List of Biers 1`] = `
<div
  className="bierlist"
>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "18" Imperial IPA 2
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          11.1
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "633" American Pale Ale
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          6.33
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "Admiral" Stache
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          7
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "Ah Me Joy" Porter
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          5.4
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "Alternating Currant" Sour
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          4.8
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "B" Street Pineapple Blonde
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          4.6
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "B.B. Rodriguez" Double Brown
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          8.5
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "Bison Eye Rye" Pale Ale | 2 of 4 Part Pale Ale Series
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          5.8
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "California Crude" Black IPA
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          7.6
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
  <div>
    <a
      className="bierCard "
      href="#"
      onClick={[Function]}
    >
      <h2>
        "Câ€™est Noir" Imperial Stout
      </h2>
      <ul
        className="bierstats"
      >
        <li>
          ABV: 
          10.8
          %
        </li>
        <li>
          Status: 
          Verified
        </li>
      </ul>
    </a>
  </div>
</div>
`;
